pkg/metadatastorage/attestationcollection/parser_registry.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2024 The Archivista Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attestationcollection
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/in-toto/archivista/ent"
+	"log"
+)
+
+var registeredParsers map[string]AttestationParser
```

Comment:
Should this functionality be up one level? Or the parsers shared outside of collections? https://github.com/in-toto/archivista/blob/main/pkg/metadatastorage/parserregistry/registry.go#L23

Reply:
I kept it in the same package so that I could make the map of parsers private to the package and prevent tampering. The only way to get something in is to use the Register() function.

We could move it further up, but we'll need to think about how this affects the access patterns.